Tests/TSCUtilityTests/BitstreamTests.swift

```diff
@@ -183,4 +183,410 @@ final class BitstreamTests: XCTestCase {
+            emitRecordID(DiagnosticRecordID.category, named: "CatName", to: recordWriter)
+            emitRecordID(DiagnosticRecordID.diagnosticFlag, named: "DiagFlag", to: recordWriter)
+            emitRecordID(DiagnosticRecordID.filename, named: "FileName", to: recordWriter)
+            emitRecordID(DiagnosticRecordID.fixIt, named: "FixIt", to: recordWriter)
+
+            let sourceLocationOperands: [Bitstream.Abbreviation.Operand] = [
+                .fixed(bitWidth: 10), // File ID
+                .fixed(bitWidth: 32), // Line
+                .fixed(bitWidth: 32), // Column
+                .fixed(bitWidth: 32), // Offset
+            ]
+
+            abbreviations.append(recordWriter.defineBlockInfoAbbreviation(.diagnostics, .init([
+                .literalCode(DiagnosticRecordID.diagnostic),
+                .fixed(bitWidth: 3)   // Diag level.
+            ] + sourceLocationOperands + [
+                .fixed(bitWidth: 10), // Category.
+                .fixed(bitWidth: 10), // Mapped Diag ID.
+                .fixed(bitWidth: 16), // Text size.
+                .blob                 // Diagnostic text.
+            ])))
+
+            abbreviations.append(recordWriter.defineBlockInfoAbbreviation(.diagnostics, .init([
+                .literalCode(DiagnosticRecordID.category),
+                .fixed(bitWidth: 16), // Category ID
+                .fixed(bitWidth: 8),  // Text size
+                .blob                 // Category text
+            ])))
+
+            abbreviations.append(recordWriter.defineBlockInfoAbbreviation(.diagnostics, .init([
+                .literalCode(DiagnosticRecordID.sourceRange)
+            ] + sourceLocationOperands + sourceLocationOperands)))
+
+            abbreviations.append(recordWriter.defineBlockInfoAbbreviation(.diagnostics, .init([
+              .literalCode(DiagnosticRecordID.diagnosticFlag),
+              .fixed(bitWidth: 10), // Mapped Diag ID
+              .fixed(bitWidth: 16), // Text size
+              .blob                 // Flag name text
+            ])))
+
+            abbreviations.append(recordWriter.defineBlockInfoAbbreviation(.diagnostics, .init([
+              .literalCode(DiagnosticRecordID.filename),
+              .fixed(bitWidth: 10), // Mapped File ID
+              .fixed(bitWidth: 32), // Size
+              .fixed(bitWidth: 32), // Modification time.
+              .fixed(bitWidth: 16), // Text size.
+              .blob                 // File name text
+            ])))
+
+            abbreviations.append(recordWriter.defineBlockInfoAbbreviation(.diagnostics, .init([
+                .literalCode(DiagnosticRecordID.fixIt)
+            ] + sourceLocationOperands + sourceLocationOperands + [
+                .fixed(bitWidth: 16), // Text size
+                .blob                // FixIt Text
+            ])))
+        }
+
+        XCTAssertEqual(recordWriter.data, [
```

Comment:
We could probably pull this test out into its own clang serializer, honestly. Later, not for this PR.